Framework/Algorithms/src/GetAllEi.cpp

```diff
@@ -478,8 +484,8 @@ bool GetAllEi::peakGuess(const API::MatrixWorkspace_sptr &inputWS, size_t index,
   double xOfMax(0), dXmax(0);
   double Intensity(0);
 
-  auto X = inputWS->readX(index);
-  auto S = inputWS->readY(index);
+  const auto &X = inputWS->x(index);
+  const auto &S = inputWS->y(index);
```

Comment:
@LamarMoore Yes, but there is also other stuff going on (like setting `dXmax`), it is would be kind of hard to replace this by `auto S = inputWS->frequencies(index)`. But maybe there is a way, I did not look in detail.